reaper.go

```diff
@@ -237,6 +257,39 @@ func newReaper(ctx context.Context, sessionID string, provider ReaperProvider, o
 
 	c, err := provider.RunContainer(ctx, req)
 	if err != nil {
+		// We need to check whether the error is caused by a container with the same name
+		// already existing due to race conditions. We manually match the error message
+		// as we do not have any error types to check against.
```

Comment:
Do you think that creating our own error here would make sense? Given the description, it seems it's needed and I think the reaper is part of our own infra, which could deserve its own error type. Thoughts?

Reply:
What do you mean exactly? The regex matching is because the returned error from the Docker client is not typed. We only get a string representation from the SDK, no error codes, etc.

Reply:
Indeed. I wonder if i't worth it to create our own error for that case, doing the regexp when asserting the error type. But probably is overkill. I wanted to check with you first

Reply:
Okay, I get it. Would definitely be more clean but if I add it for the `DockerProvider` and only check against this error type, other implementations of `ContainerProvider` that people might use, will definitely not work with this as they would need to update their implementations to return the correct error, right?

Reply:
I had in mind a custom error type, something like `ErrReuseEmptyName = errors.New("with reuse option a container name mustn't be empty")`, not added to the Provider.

But in any case, let's not block this PR and keep it simple :)

Reply:
Mhm, but for what would the error be? I assumed you mean a custom error that is returned by `RunContainer` (or underlying  methods). This would allow replacing the regex check in _reaper.go_ with `errors.Is`. But then we would expect the Provider to actually return this custom error. And custom implementations of `ContainerProvider` would need to return the correct type as well for this to work.

Reply:
I was referring defining our own error type, and creating a type assertion for the error, something like:

```go
type duplicatedReaperError struct
   Message string
}

func NewDuplicatedReaperError(message string) error {
	return &duplicatedReaperError{message}
}

func (err *duplicatedReaperError) Error() string {
   return err.Message
}

// IsDuplicatedReaperError checks if an error of type duplicatedReaperError exists.
func IsDuplicatedReaperError(err error) bool {
	var e duplicatedReaperError
	// do the regex here
	return errors.As(err, &e)
}
```

but in any case, I'm not forcing it :D It's fine as it is now.